tests/regression/test_interpolate.py

```diff
@@ -0,0 +1,121 @@
+import pytest
+import numpy as np
+from firedrake import *
+from tests.common import *
+
+
+def test_constant(cg1):
+    f = interpolate(Constant(1.0), cg1)
+    assert np.allclose(1.0, f.dat.data)
+
+
+def test_function():
+    m = UnitTriangleMesh()
+    V1 = FunctionSpace(m, 'P', 1)
+    V2 = FunctionSpace(m, 'P', 2)
+
+    f = interpolate(Expression("x[0]*x[0]"), V1)
+    g = interpolate(f, V2)
+
+    # g shall be equivalent to:
+    h = interpolate(Expression("x[0]"), V2)
```

Comment:
This works because the interpolation points for f are the vertices of the triangle.